ui/src/timeMachine/components/SelectorListCreateBucket.tsx

```diff
@@ -0,0 +1,203 @@
+// Libraries
+import React, {
+  FC,
+  ChangeEvent,
+  FormEvent,
+  useState,
+  useEffect,
+  useRef,
+} from 'react'
+import {connect} from 'react-redux'
+
+// Components
+import {
+  Popover,
+  PopoverInteraction,
+  PopoverPosition,
+  Appearance,
+  ComponentColor,
+} from '@influxdata/clockface'
+import BucketOverlayForm from 'src/buckets/components/BucketOverlayForm'
+
+// Utils
+import {
+  extractBucketMaxRetentionSeconds,
+  extractBucketLimits,
+} from 'src/cloud/utils/limits'
+
+// Constants
+import {
+  DEFAULT_SECONDS,
+  READABLE_DEFAULT_SECONDS,
+} from 'src/buckets/components/Retention'
+
+// Actions
+import {
+  checkBucketLimits as checkBucketLimitsAction,
+  LimitStatus,
+} from 'src/cloud/actions/limits'
+import {createBucket} from 'src/buckets/actions/thunks'
+
+// Types
+import {Organization, Bucket, AppState} from 'src/types'
+
+// Selectors
+import {getOrg} from 'src/organizations/selectors'
+
+const DEFAULT_RULES = [
+  {type: 'expire' as 'expire', everySeconds: DEFAULT_SECONDS},
+]
+
+interface StateProps {
+  org: Organization
+  isRetentionLimitEnforced: boolean
+  limitStatus: LimitStatus
+}
+
+interface DispatchProps {
+  createBucket: typeof createBucket
+  checkBucketLimits: typeof checkBucketLimitsAction
+}
+
+interface OwnProps {}
+
+type Props = OwnProps & StateProps & DispatchProps
+
+const SelectorListCreateBucket: FC<Props> = ({
+  org,
+  createBucket,
+  isRetentionLimitEnforced,
+  limitStatus,
+  checkBucketLimits,
+}) => {
+  const triggerRef = useRef<HTMLButtonElement>(null)
+  const [bucketName, setBucketName] = useState<string>('')
+  const [bucketRetentionRules, setBucketRetentionRules] = useState<any>(
+    isRetentionLimitEnforced ? DEFAULT_RULES : []
+  )
+  const [bucketRuleType, setBucketRuleType] = useState<'expire' | null>(
```

Comment:
There's enough state in here to justify moving this into a `useReducer` hook.   It could all even stay in this file but would pull a bunch of this state management into a reducer.

Reply:
A similar thought I had was to make a component that is just the contents of the popover, then most of the state can live in there. We'd still have a lot of state though. The component I based this off of just had a whole `Bucket` object as a single piece of state so I flattened it out